test/e2e/suite_test.go

```diff
@@ -173,6 +176,11 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 	framework.InitClusterInformation(karmadaClient, controlPlaneClient)
 
+	controllerManager = framework.InitControllerManagerAndStartCache(restConfig)
+
+	err = utilhelper.IndexWork(context.TODO(), controllerManager)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
```

Comment:
Could you please remind me where referencing the controllerManager?

In my opinion, the E2E tests don't need to build controllers themselves.

Reply:
https://github.com/karmada-io/karmada/blob/20af47da0b968bd83cae482c0017f1346071a036/test/e2e/clusterresourcebinding_test.go#L83-L86

any suggestions for that case?

Reply:
cc @XiShanYongYe-Chang to look at it.

Reply:
Let me take a look.

> In my opinion, the E2E tests don't need to build controllers themselves.

I agree with this point.

I have looked at the E2E test content. Perhaps our case check is not end-to-end enough. After all, Work is a product in the middle of the system.

In addition, if the case needs to be retained, the GetWorksByBindingID method invoked by the case can be maintained separately in the E2E without introducing a controllerManager in the E2E because the original invoked function changes.

For the current case, my suggestion is that we delete it first. What do you think?

Reply:
Hi @zach593 do you need me to bring up a pr to clean up?

Reply:
If the direction is decoupled from this helper function, I think we can use `helper.GetWorksByLabelsSet()` here. This way is also very simple, and it maintains the original effect.